.github/workflows/build.yml

```diff
@@ -62,6 +84,28 @@ jobs:
       - name: Build
         run: make
 
+      - name: Dry run test with VIP generation
+        shell: bash
+        run: |
+          pushd . > /dev/null
+          cd .github/dry-run-test
+          ./generate_flat_build.sh
+          mkdir -p vip
+          ../../qdl --debug --dry-run --create-digests=./vip prog_firehose_ddr.elf rawprogram*.xml patch*.xml
+          actual_digest=$(shasum -a 256 "./vip/DigestsToSign.bin" | cut -d ' ' -f1)
+          expected_digest="4e13981189ede380172369aa0a4847ca3c9a4e2795733bca90ec1b1e713972ea"
+          popd > /dev/null
+          if [ "$actual_digest" = "$expected_digest" ]; then
+            echo "VIP tables are generated successfully and validated"
+            exit 0
+          else
+            echo "Expected SHA256 digest of DigestsToSign.bin file is $expected_digest"
+            echo "Calculated SHA256 digest of DigestsToSign.bin file is $actual_digest"
+            echo "VIP table folder contents:"
+            ls -la .github/dry-run-test/vip
+            exit 1
+          fi
```

Comment:
Could you please extract this to a script, removing unnecessary duplication between workflow stages.